File: src/main/kotlin/ru/quipy/logic/UserAggregateState.java
package ru.quipy.logic;

import kotlin.UninitializedPropertyAccessException;
import lombok.AllArgsConstructor;
import lombok.Getter;
import ru.quipy.api.UserAggregate;
import ru.quipy.domain.AggregateState;

import java.util.UUID;

@Getter
@AllArgsConstructor
public class UserAggregateState implements AggregateState<UUID, UserAggregate> {
    private UUID id;
    private String username;
    private String password;
    private String fio;

    @Override
    public UUID getId() {
        if (id == null) {
            throw new UninitializedPropertyAccessException("Property has not been initialized");
        }
        return id;
    }
}

File: src/main/kotlin/ru/quipy/logic/UserAggregateState.kt
package ru.quipy.logic

import ru.quipy.api.UserAggregate
import ru.quipy.api.UserRegisteredEvent
import ru.quipy.core.annotations.StateTransitionFunc
import ru.quipy.domain.AggregateState
import java.util.*

class UserAggregateState : AggregateState<UUID, UserAggregate> {
    private lateinit var userId: UUID
    var createdAt: Long = System.currentTimeMillis()
    var updatedAt: Long = System.currentTimeMillis()

    lateinit var username: String
    lateinit var password: String
    lateinit var fio: String
    override fun getId() = userId

    @StateTransitionFunc
    fun userRegisteredApply(event: UserRegisteredEvent){
        userId = event.userId
        username = event.username
        password = event.password
        updatedAt = createdAt
    }
}